server/bastionlab_common/src/common_conversions.rs

```diff
@@ -0,0 +1,315 @@
+                d
+            )))
+        }
+    })
+}
+
+pub fn lazy_frame_from_logical_plan(plan: LogicalPlan) -> LazyFrame {
+    let mut ldf = LazyFrame::default();
+    ldf.logical_plan = plan;
+    ldf
+}
+
+pub fn to_status_error<T, E: Error>(input: Result<T, E>) -> Result<T, Status> {
+    input.map_err(|err| Status::aborted(err.to_string()))
+}
+
+pub fn load_udf(udf: String) -> Result<CModule, Status> {
+    Ok(
+        CModule::load_data(&mut Cursor::new(base64::decode(udf).map_err(|e| {
+            Status::invalid_argument(format!("Could not decode bas64-encoded udf: {}", e))
+        })?))
+        .map_err(|e| {
+            Status::invalid_argument(format!("Could not deserialize udf from bytes: {}", e))
+        })?,
+    )
+}
+
+fn get_tokenizer(model: &str) -> Result<Tokenizer, Status> {
+    let model = base64::decode_config(model, base64::STANDARD).map_err(|e| {
+        Status::invalid_argument(format!("Could not decode bas64-encoded udf: {}", e))
+    })?;
+    let tokenizer: Tokenizer = Tokenizer::from_bytes(model)
+        .map_err(|_| Status::invalid_argument("Could not deserialize Hugging Face Tokenizer"))?;
+    Ok(tokenizer)
+}
+
+pub fn series_to_tokenized_series(
+    s: &Series,
+    name: &str,
+    model: &str,
+) -> Result<DataFrame, Status> {
+    let tokenizer = get_tokenizer(model)?;
+    let mut batched_seqs = Vec::new();
+
+    let to_row = |tokens: &Encoding| {
+        let ids = tokens.get_ids();
+        let mask = tokens.get_attention_mask();
+
+        let to_any_value = |v: &[u32]| -> AnyValue {
+            let mut buf = AnyValueBuffer::new(&DataType::UInt32, v.len());
+            v.iter().for_each(|v| {
+                buf.add(AnyValue::UInt32(*v));
+            });
+            AnyValue::List(buf.into_series())
+        };
+        let ids = to_any_value(ids);
+        let mask = to_any_value(mask);
+
+        let joined = vec![ids.clone(), mask.clone()];
+
+        let row = Row::new(joined);
+        row
+    };
+    for row in s.utf8().unwrap().into_iter() {
+        match row {
+            Some(s) => {
+                batched_seqs.push(s.to_string());
+            }
+            None => {
+                return Err(Status::aborted(
+                    "Failed to convert row to Utf8 string".to_string(),
+                ));
+            }
+        }
+    }
+
+    let tokens_vec = tokenizer
+        .encode_batch(batched_seqs, false)
+        .map_err(|_| Status::aborted("Failed to tokenize string"))?;
+
+    let rows = tokens_vec.iter().map(to_row).collect::<Vec<_>>();
+    let mut df = to_status_error(DataFrame::from_rows(&rows[..]))?;
+
+    let ids_names = &format!("{}_ids", name.to_lowercase());
+    let mask_names = &format!("{}_mask", name.to_lowercase());
+
+    let col_names = df.get_column_names_owned();
+    to_status_error(df.rename(&col_names[0], &ids_names))?;
+    to_status_error(df.rename(&col_names[1], &mask_names))?;
+    Ok(df)
+}
+
+#[allow(unused)]
```

Comment:
???

Reply:
This can be taken out for now.
It'll be useful when we do conversion from a tokenized series into strings.